conda_declarative/spec.py

```diff
@@ -0,0 +1,821 @@
+from __future__ import annotations
+
+import string
+from abc import ABC, abstractmethod
+from collections.abc import Iterable
+from copy import copy
+from dataclasses import fields
+from functools import total_ordering
+from pathlib import Path
+from pprint import pformat
+from textwrap import indent
+from typing import Annotated, Any
+from warnings import warn
+
+try:
+    from tomllib import loads
+except ImportError:
+    from tomli import loads
+
+from conda.base.context import context
+from conda.models.environment import Environment, EnvironmentConfig
+from conda.models.match_spec import MatchSpec
+from conda.models.records import PrefixRecord
+from conda.plugins.types import EnvironmentSpecBase
+from pydantic import (
+    AnyHttpUrl,
+    BaseModel,
+    BeforeValidator,
+    ConfigDict,
+    EmailStr,
+    PlainSerializer,
+    RootModel,
+    ValidationError,
+    ValidationInfo,
+    field_validator,
+    model_serializer,
+    model_validator,
+)
+
+
+def validate_match_spec(
+    value: list[MatchSpec | EditablePackage]
```

Comment:
It looks like this value can be a list of a dict.
```suggestion
    value: Iterable[MatchSpec | EditablePackage]
```